pkg/oceancd/model/operator/install.go

```diff
@@ -0,0 +1,171 @@
+package operator
+
+import (
+	"encoding/json"
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	"spot-oceancd-operator-commons/component_configs"
+)
+
+type InstallationConfig struct {
+	OceanCDConfig      OceanCDConfig                        `json:"oceancd"`
+	ArgoRolloutsConfig component_configs.ArgoRolloutsConfig `json:"argo"`
+}
+
+func (c *InstallationConfig) GetData() map[string]interface{} {
+	bytes, _ := json.Marshal(c)
+
+	data := map[string]interface{}{}
+	_ = json.Unmarshal(bytes, &data)
+
+	return data
+}
+
+func (c *InstallationConfig) GetOperatorManagerConfig() *component_configs.OperatorManagerConfig {
+	omConfig := &component_configs.OperatorManagerConfig{
+		ArgoRolloutsConfig: c.ArgoRolloutsConfig,
+		OceanCDConfig: component_configs.OceanCDConfig{
+			Operator: c.OceanCDConfig.Operator,
+		},
+	}
+
+	omConfig.OceanCDConfig.Operator.Namespace = c.OceanCDConfig.Namespace
+
+	return omConfig
+}
+
+type OceanCDConfig struct {
+	Namespace     string                                  `json:"namespace"`
+	ManagerConfig ManagerConfig                           `json:"manager"`
+	Operator      component_configs.OceanCDOperatorConfig `json:"operator"`
+}
+
+type ManagerConfig struct {
+	PodLabels        map[string]string             `json:"podLabels"`
+	PodAnnotations   map[string]string             `json:"podAnnotations"`
+	Labels           map[string]string             `json:"labels"`
+	NodeSelector     map[string]string             `json:"nodeSelector"`
+	Affinity         corev1.Affinity               `json:"affinity"`
+	Tolerations      []corev1.Toleration           `json:"tolerations"`
+	ExtraEnv         []corev1.EnvVar               `json:"extraEnv"`
+	Resources        corev1.ResourceRequirements   `json:"resources"`
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets"`
+}
+
+func NewInstallationConfig(data map[string]interface{}) (*InstallationConfig, error) {
+	dataBytes, err := json.Marshal(data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal installation config: %w", err)
+	}
+
+	config := DefaultInstallationConfig()
+
+	if err := json.Unmarshal(dataBytes, &config); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal installation config: %w", err)
+	}
+
+	return &config, nil
+}
+
+func DefaultInstallationConfig() InstallationConfig {
```

Comment:
Change to DefaultOperatorManagerConfig

Reply:
we already have OperatorManagerConfigthat stores argoRollouts and operator's configs that then converted into ConfigMap

the InstallationConfig also contains info about the operator manager, the config is used only for the installation of OM

I suggest leave it as it is